2_recipe_batches/recipe_batches.py

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/python
+
+import math
+
+recipe = { 'milk': 100, 'butter': 50, 'flour': 5 }
+ingredients = { 'milk': 132, 'butter': 48, 'flour': 51 }
+
+print(ingredients['milk']/recipe['milk'])
+print(ingredients['butter']/recipe['butter'])
+print(ingredients['flour']/recipe['flour'])
+
```

Comment:
As for `for...in...` loop over objects, when you do `for key in recipe`, the value of `key` in each loop will be a key/property name in the dictionary. So,
```
for key in recipe:
    print(key)
# output -> `milk` `butter` `flour`
```
(_Remember, `key` is a variable name. You can name it anything, but it will have the value of the current property name of an object in any given loop_). 

Does this give you ideas on refactoring your current function?

Reply:
Yes.  I think I just needed to try a few operations to see this.